src/org/zaproxy/zap/extension/jsonView/ExtensionHttpPanelJsonView.java

```diff
@@ -0,0 +1,213 @@
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.jsonView;
+
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.httppanel.Message;
+import org.zaproxy.zap.extension.httppanel.component.split.response.ResponseSplitComponent;
+import org.zaproxy.zap.extension.httppanel.component.split.request.RequestSplitComponent;
+import org.zaproxy.zap.extension.httppanel.view.HttpPanelDefaultViewSelector;
+import org.zaproxy.zap.extension.httppanel.view.HttpPanelView;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.request.RequestBodyStringHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.response.ResponseBodyStringHttpPanelViewModel;
+import org.zaproxy.zap.view.HttpPanelManager;
+import org.zaproxy.zap.view.HttpPanelManager.HttpPanelDefaultViewSelectorFactory;
+import org.zaproxy.zap.view.HttpPanelManager.HttpPanelViewFactory;
+
+public class ExtensionHttpPanelJsonView extends ExtensionAdaptor {
+	
+	public static final String NAME = "ExtensionHttpPanelJsonView";
+
+	@Override
+	public String getAuthor() {
+		return "Juha Kivekäs";
+	}
+
+	public ExtensionHttpPanelJsonView() {
+		super(NAME);
+	}
+
+	@Override
+	public void init() {
+		super.init();
+	}
+
+	@Override
+	public void hook(ExtensionHook extensionHook) {
+	    super.hook(extensionHook);
+
+		if (getView() != null) {
+			HttpPanelManager panelManager = HttpPanelManager.getInstance();
+			//add the view factories for http requests and responses
+			panelManager.addRequestViewFactory(RequestSplitComponent.NAME, new RequestJsonViewFactory());
+			panelManager.addResponseViewFactory(ResponseSplitComponent.NAME, new ResponseJsonViewFactory());
+			//add default view selectors
+			panelManager.addRequestDefaultViewSelectorFactory(RequestSplitComponent.NAME,
+					new JsonDefaultViewSelectorFactory(RequestSplitComponent.NAME, RequestSplitComponent.ViewComponent.BODY));
+			panelManager.addResponseDefaultViewSelectorFactory(ResponseSplitComponent.NAME,
+					new JsonDefaultViewSelectorFactory(ResponseSplitComponent.NAME, ResponseSplitComponent.ViewComponent.BODY));
+		}
+	}
+	
+	@Override
+	public boolean canUnload() {
+		return true;
+	}
+
+	@Override
+	public void start() {
+		super.start();
+	}
+	
+	@Override
+	public void unload() {
+		super.unload();
+		HttpPanelManager panelManager = HttpPanelManager.getInstance();

```

Comment:
Should be wrapped in a `if (getView() != null) {` (like in the `hook`).

Reply:
check